src/compiler/checker.ts

```diff
@@ -2595,7 +2595,14 @@ namespace ts {
 
             // Use the type of the initializer expression if one is present
             if (declaration.initializer) {
-                return checkExpressionCached(declaration.initializer);
+                let mapper: TypeMapper;
+                if (declaration.kind === SyntaxKind.PropertyDeclaration) {
+                    const type = getTypeOfBasePropertyDeclaration(<PropertyDeclaration>declaration);
+                    if (type) {
+                        mapper = createTypeMapper([undefinedType, nullType], [type, type]);
+                    }
+                }
+                return checkExpressionCached(declaration.initializer, mapper);

```

Comment:
The `contextualMapper` parameter isn't intended to be used this way. The comment on `checkExpression` explains its purpose.